db/migrations/2402201446_new_sessions_schema.sql

```diff
@@ -0,0 +1,135 @@
+# INSERT INTO `sessions` (`access_token`, `user_id`, `expires_at`, `issued_at`, `issuer`) VALUES ('access_token_2', 2, '2024-01-01 02:00:00', '2024-01-01 01:00:00', 'login-module');
+# INSERT INTO `sessions` (`access_token`, `user_id`, `expires_at`, `issued_at`, `issuer`) VALUES ('access_token_3', 3, '2024-01-01 02:00:00', '2024-01-01 01:00:00', 'backend');
+# INSERT INTO `sessions` (`access_token`, `user_id`, `expires_at`, `issued_at`, `issuer`) VALUES ('access_token_4', 4, '2024-01-01 02:00:00', '2024-01-01 01:00:00', 'backend');
+#
+# The result should be:
+#
+# SELECT * FROM `sessions`;
+# +------------+---------+----------------+
+# | session_id | user_id | refresh_token  |
+# +------------+---------+----------------+
+# | 1          | 1       | refresh_token_1|
+# | 2          | 2       | refresh_token_2|
+# | 3          | 3       | NULL           |
+# | 4          | 4       | NULL           |
+# +------------+---------+----------------+
+#
+# SELECT * FROM `access_tokens`;
+# +-----------------+------------+---------------------+---------------------+
+# | token           | session_id | expires_at          | issued_at           |
+# +-----------------+------------+---------------------+---------------------+
+# | access_token_1a | 1          | 2024-01-01 01:00:00 | 2024-01-01 00:00:00 |
+# | access_token_1b | 1          | 2024-01-01 02:00:00 | 2024-01-01 01:00:00 |
+# | access_token_2  | 2          | 2024-01-01 00:00:00 | 2024-01-01 01:00:00 |
+# | access_token_3  | 3          | 2024-01-01 00:00:00 | 2024-01-01 01:00:00 |
+# | access_token_4  | 4          | 2024-01-01 00:00:00 | 2024-01-01 01:00:00 |
+# +----------------+------------+---------------------+---------------------+
+#
+
+
+CREATE TABLE `new_sessions` (
+  `session_id` BIGINT NOT NULL AUTO_INCREMENT,
+  `user_id` BIGINT NOT NULL,
+  CONSTRAINT `fk_sessions_users_user_id_group_id` FOREIGN KEY (`user_id`) REFERENCES `users`(`group_id`) ON DELETE CASCADE,
+  `refresh_token` VARBINARY(2000) NULL COMMENT 'Refresh tokens (unlimited lifetime) used by the backend to request fresh access tokens from the auth module',
+  PRIMARY KEY (`session_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COMMENT='Sessions represent a logged in user, on a specific device.';
+
+CREATE TABLE `access_tokens` (
+  `token` VARBINARY(2000) NOT NULL COMMENT 'The access token.',
+  `session_id` BIGINT NOT NULL,
+  CONSTRAINT `fk_access_tokens_sessions_session_id` FOREIGN KEY (`session_id`) REFERENCES `new_sessions`(`session_id`) ON DELETE CASCADE,
+  `expires_at` DATETIME NOT NULL COMMENT 'The time the token expires and becomes invalid. It should be deleted after this time.',
+  `issued_at` DATETIME NOT NULL DEFAULT NOW() COMMENT 'The time the token was issued.'
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COMMENT='Access tokens (short lifetime) distributed to users, to access a specific session.';
+
```

Comment:
I see you lost most of the indexes that were existing before: (at least in this commit)
```
 INDEX `expires_at` (`expires_at`) USING BTREE,
  INDEX `access_token_prefix` (`access_token`(767)) USING BTREE,
  INDEX `user_id` (`user_id`) USING BTREE,
  INDEX `refresh_token_prefix` (`refresh_token`(767)) USING BTREE,
```
I assume most of them were making sense. (including this way of indexing the prefix)

Reply:
fixed. Added the same indexes.

From what I could see, only the index on `access_token` is really useful. The other ones I'm not sure, but it shouldn't hurt too much as those tables have few indexes anyway.